packages/in_app_purchase/example/lib/main.dart

```diff
@@ -67,14 +69,30 @@ class _MyAppState extends State<MyApp> {
       return;
     }
 
-    ProductDetailsResponse productDetails =
-        await _connection.queryProductDetails(_kProductIds.toSet());
-    if (productDetails.productDetails.isEmpty) {
+    ProductDetailsResponse productDetailResponse;
+    try {
+      productDetailResponse =
+          await _connection.queryProductDetails(_kProductIds.toSet());
+    } on PlatformException catch (_) {
```

Comment:
Is this also throwing in airplane mode?

If so, I'm thinking maybe we should this to also populate an error in the response for cases like this. I'd rather not expect people to have to `try / catch` around the API for standard use cases like the phone being offline, if possible.